app/assets/javascripts/hqmf_util.js.coffee

```diff
@@ -875,7 +875,9 @@ eventsMatchBounds = (events, bounds, methodName, range) ->
     events = [events]
 
   specificContext = new hqmf.SpecificOccurrence()
-  hasSpecificOccurrence = (events.specific_occurrence? || bounds.specific_occurrence?)
+  # For the bounds (RHS), we check not only if the immediate RHS has specifics, but also whether anything on
+  # the RHS has specifics steps further removed, by checking if there's a specificContext with specifics
+  hasSpecificOccurrence = (events.specific_occurrence? || bounds.specific_occurrence? || bounds.specificContext?.hasSpecifics())
```

Comment:
I think that, because of how QDM logic chains (and the lack of parentheses), the LHS can't be "steps removed" from where the specifics occur.